### PR TITLE
Update DeveloperGuide and UserGuide

### PR DESCRIPTION
**Pull request by xiaoge26**
1. Fixed some typos in the DeveloperGuide.md and standardized the format (Uppercase /lowercase issues, no full stop at the end of a bullet point etc.)
2. Added Setting Up, Getting Started, Non-Functional Requirements and Instructions for Manual Testing with reference to [AddressBoook-Level3](https://se-education.org/addressbook-level3/DeveloperGuide.html).
3. Fixed some mistakes and outdated commands in the UserGuide.md, but there are still some left. You may go and check it again.
4. Added PPP for myself as well!

Note:  DeveloperGuide.md may need further improvements. Some outdated/unfinished parts may still exist in the UserGuide.md. Now we have 14 commands, but not all of them have been included in the guide. Hence, please do check the guides and feel free to add more.

### CODE REVIEW

**Comment by xiaoge26** [comment]
This PR can be reviewed after PR #159 is merged. Thank you!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -169,26 +176,42 @@ minimalism of a CLI interface, as it provides only the essential information and
...
-| v1.0    | user     | view account                             | so that i can keep track of my minimum balance                                          |
-| v2.0    | user     | add a withdrawal limit                   | so that i can be forced to maintain a frugal habit                                      |
-| v2.0    | user     | add a savings goal                       | so that i can keep track of my spendings in tandem with my savings goal                 |
-| v2.0    | user     | add a new account                        | so that i can categorise my accounts and manage financial better                        |
-| v2.0    | user     | switch account                           | so that i can navigate between the various accounts i have and keep track of finacnials |
-| v2.0    | user     | list all the command that can be used    | so that i can have a basic overview of the apps functionalities                         |
+| Version | As a ... | I want to ...                         | So that I can ...                                                                       |
```
Comment (manushridiv) [inline docs/DeveloperGuide.md:179]: formatting is better though not much changes from the initital start

File: docs/README.md
Comment (xiaoge26) [inline docs/README.md:1]: I accidentally edited README.md.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 1. Ensure you have Java `11` or above installed in your Computer.
 
-2. Download the latest `BankWithUs.jar` from [here](https://github.com/AY2223S2-CS2113-T13-3/tp/releases).
+2. Download the latest `tp.jar` from [here](https://github.com/AY2223S2-CS2113-T13-3/tp/releases).
```
Comment (vishnuvk47) [inline docs/UserGuide.md:9]: I think might be better to leave it as BankWithUs.jar in the UG and instead change the name of the jar file to BankWithUs before release.

**Review by vishnuvk47 — APPROVED** [review]
LGTM! Can merge for now, name of .jar file in UG can change later.

File: docs/UserGuide.md (hunk 2 above)
Comment (xiaoge26) [reply docs/UserGuide.md:9]: Ok sure!

Comment (xiaoge26) [reply docs/UserGuide.md]: I changed it back to BankWithUs.jar just now.

**Pull request merged**